File: map/map.js
// To run locally, use: node map.js

// this is a silly example, but it's a good example of how to use the map function
const arr = [
  { displayName: 'John', familyName: 'Doe' },
  { displayName: 'Jane', familyName: 'Doe' },
  { displayName: 'Jaken', familyName: 'Herman' }
];

let res = arr.map(r => ({
  name: r.displayName,
  family: r.familyName
}));

console.log(res); 

/** ^ ^ ^ ^ ^ ^   
[
  { name: 'John', family: 'Doe' },
  { name: 'Jane', family: 'Doe' },
  { name: 'Jaken', family: 'Herman' }
]  
*/

File: map/map.rs
// To run locally, use: rustc map.rs && ./map

#[derive(Debug)]
struct Person {
    display_name: String,
    family_name: String,
}

#[derive(Debug)]
struct SimplePerson {
    name: String,
    family: String,
}

fn main() {

    // this is a silly example, but it's a good example of how to use the map function
    let arr = [
        Person { display_name: String::from("John"), family_name: String::from("Doe") },
        Person { display_name: String::from("Jane"), family_name: String::from("Doe") },
        Person { display_name: String::from("Jaken"), family_name: String::from("Herman") },
    ];


    let res = arr.map(|r| SimplePerson {
      name: r.display_name.clone(),
      family: r.family_name.clone(),
    });
    println!("{:?}", res);
    // ^^^
    // [
        // SimplePerson { name: "John", family: "Doe" },
        // SimplePerson { name: "Jane", family: "Doe" },
        // SimplePerson { name: "Jaken", family: "Herman" }
    // ]
